DMCompiler/Compiler/DM/DMParser.cs

```diff
@@ -592,7 +592,26 @@ public DMASTProcStatement ProcStatement()
                 }
 
                 return varDeclarations.ToArray();
-            } else if (hasNewline) {
+            } else if (Check(TokenType.DM_LeftCurlyBracket)) {
+                Whitespace();
+                Newline();
+                bool isIndented = Check(TokenType.DM_Indent);
+
+                List<DMASTProcStatementVarDeclaration> varDeclarations = new();
+                while (!Check(isIndented ? TokenType.DM_Dedent : TokenType.DM_RightCurlyBracket)) {
+                    DMASTProcStatementVarDeclaration[] varDecl = ProcVarEnd(true, path: varPath);
+                    Check(TokenType.DM_Semicolon);
+                    if (varDecl == null) Error("Expected a var declaration");
+
+                    varDeclarations.AddRange(varDecl);
+                    Newline();
```

Comment:
why not also check for whitespace here? ex.
```
var/{flatX1=1; flatX2=2}
```

Reply:
```suggestion
                    Whitespace();
                    Newline();
```